src/server/utils/settings-manager/settings-manager.ts

```diff
@@ -156,40 +155,43 @@ export class SettingsManager {
     return this.timeMonitor.timekeeper;
   }
 
-  getSettings(opts: GetSettingsOptions = {}): Q.Promise<AppSettings> {
-    var currentWork = this.currentWork;
-
+  getSettings(opts: GetSettingsOptions = {}): Promise<AppSettings> {
     // Refresh all clusters
-    var currentWork = currentWork.then(() => {
+    var currentWork = this.currentWork.then(() => {
       // ToDo: utilize dataCubeOfInterest
-      return Q.all(this.clusterManagers.map(clusterManager => clusterManager.refresh())) as any;
+      return Promise.all(this.clusterManagers.map(clusterManager => clusterManager.refresh())) as any;
     });
 
     var timeout = opts.timeout || this.initialLoadTimeout;
     if (timeout !== 0) {
-      currentWork = currentWork.timeout(timeout)
-        .catch(e => {
-          this.logger.error("Settings load timeout hit, continuing");
-        });
+      currentWork = Promise.race([
+        currentWork,
+        new Promise((resolve, reject) => {
```

Comment:
Could be usefull helper `timeout` which returns Promise that would resolve after x ms.